mailbox/backup/src/main/java/org/apache/james/mailbox/backup/Backup.java

```diff
@@ -33,5 +35,5 @@
      * @param messages a stream of MailboxMessages that will be consumed
      * @param destination an OutputStream in which the zip will be written
      */
-    void archive(List<Mailbox> mailboxes, Stream<MailboxMessage> messages, OutputStream destination) throws IOException;
+    public void archive(Map<Mailbox, List<MailboxAnnotation>> mailboxes, Stream<MailboxMessage> messages, OutputStream destination) throws IOException ;
```

Comment:
Isn't `Map<Mailbox, List<MailboxAnnotation>>` better represented as a `Multimap<Mailbox, MailboxAnnotation>` ?

Reply:
It was my fist thought, but we want to serialize a Mailbox without MailBoxAnnotation and with a multimap the key is removed if there is no values.

Reply:
You probably should not rely on technical types too much.
What about having a MailboxWithAnnotation type that gather both information?
That way, you don't have to deal with Map or Multimap specific behavior and you won't have to lookup the map, neither.

Reply:
I agree with you.